java/http/java-http-producer/src/main/java/HttpProducer.java

```diff
@@ -50,12 +53,16 @@ public static void main(String[] args) throws InterruptedException, URISyntaxExc
             }
         }
 
-        HttpProducer httpProducer = new HttpProducer(config);
+        HttpProducer httpProducer = new HttpProducer(config, messagesSentLatch);
         httpProducer.run();
+
+        log.info("Waiting for sending all messages");
+        messagesSentLatch.await();
     }
 
-    public HttpProducer(HttpProducerConfig config) throws URISyntaxException {
+    public HttpProducer(HttpProducerConfig config,CountDownLatch messagesSentLatch ) throws URISyntaxException {
```

Comment:
```suggestion
    public HttpProducer(HttpProducerConfig config, CountDownLatch messagesSentLatch) throws URISyntaxException {
```

Minor typos here which can cause checkstyle to be unhappy sometimes